pages/api/v1/team.ts

```diff
@@ -1,10 +1,130 @@
+  switch (req.method) {
+    case 'GET':
+      await getHandler(req, res);
+      break;
+    case 'POST':
+      await postHandler(req, res);
+      break;
+    case 'DELETE':
+      await deleteHandler(req, res);
+      break;
+    default:
+      return res
+        .status(405)
+        .setHeader('Allow', 'GET, POST, DELETE')
+        .send(undefined);
+  }
+};
+
+const getHandler: NextApiHandler = async (req, res) => {
+  const teamName = req.query.teamName;
+  const { teamDataCollection } = await connectToDatabase();
+
+  try {
+    // Checks if the team exists
+    const existingTeam = await teamDataCollection.findOne({
+      name: teamName,
+    });
+
+    if (existingTeam != null) {
+      return res.status(200).json(existingTeam);
+    } else {
+      // Safely gets response
+      return res.status(200).json([]);
+    }
+  } catch (e) {
+    return res.status(400).send({ error: 'Error getting team information' });
+  }
+};
+
+const postHandler: NextApiHandler = async (req, res) => {
+  const { teamName, email } = req.body;
+  const { userDataCollection, teamDataCollection } = await connectToDatabase();
+
+  try {
+    await userDataCollection.updateOne(
+      { email },
+      {
+        $set: {
+          teamName,
+        },
+      },
+      { upsert: true }
+    );
+
+    const existingTeam = await teamDataCollection.findOne({
+      name: teamName,
+    });
+
+    if (existingTeam != null) {
+      // Adds the current email to team if there is a team
+      await teamDataCollection.updateOne(
+        { name: teamName },
+        {
+          $addToSet: { userEmails: email },
+        }
+      );
+    } else {
+      // Creates a team if it doesn't previously exist
+      await teamDataCollection.insertOne({
+        name: teamName,
+        userEmails: [email],
+      });
+    }
+
+    return res.status(201).send(undefined);
+  } catch (e) {
+    return res.status(400).json(e);
+  }
+};
+
+const deleteHandler: NextApiHandler = async (req, res) => {
+  const { teamName, email } = req.body;
+  const { userDataCollection, teamDataCollection } = await connectToDatabase();
+  try {
+    // Removes team name from the user data
+    await userDataCollection.updateOne(
+      {
+        email,
+      },
+      { $unset: { teamName: '' } }
+    );
+
+    const existingTeam = await teamDataCollection.findOne({
+      name: teamName,
+    });
+
+    if (existingTeam != null) {
+      // If the current person is the last person in the team, remove the whole object
+      if (
+        existingTeam.userEmails.length === 1 &&
+        existingTeam.userEmails.includes(email)
+      ) {
+        await teamDataCollection.deleteOne({
+          name: teamName,
+        });
+      } else {
+        // Else removes the current person from the team
+        await teamDataCollection.updateOne(
+          {
+            name: teamName,
+          },
+          {
+            $pull: { userEmails: email },
+          }
+        );
+      }
```

Comment:
same w this, may be able to combine api calls

Reply:
i am not sure how we can combine these since one is just removing one value from a list and the other one destroys it ;-;